Automattic/pocket-casts-ios
# Add and configure SwiftLint

This PR adds SwiftLint, but to do so, it first updates SwiftFormat and centralizes the Swift version definition in the Xcode project. These three steps could have been dedicated PRs but they ended up into such a small diff that it made sense to leave them in a single place.

Our [standard SwiftLint configuration](https://github.com/Automattic/swiftlint-config) picked up a lot of issues, the majority of them were trailing whitespaces. I separated fixing them in #283 .

**⚠️ This PR doesn't address the question of whether having _both_ SwiftLint and SwiftFormat is a good idea! ⚠️**

I was interested in implementing side-by-side support to get an idea of the work involved. From my experience here, I'd say running both tools is redundant and will only lead to issues when their default configurations diverge (see the many force pushes to this and the autocorrect application branch).

I think we should stick with SwiftLint only, because of the remote configuration with overrides support, but keen to hear what other folks think. Removing SwiftFormat would be straightforward and can be done in a dedicated PR later on.

## To test

Run `make swiftformat` and verify nothing changes. That is, the changes haven't broken or affected its setup. _CI already does this for us._

## Checklist
